### PR TITLE
feat: use codeartifact as npm cache

### PR DESCRIPTION
**Pull request by brynjarorng**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: infra/scripts/Dockerfile [hunk 1]
```diff
@@ -10,6 +10,9 @@ ENV NODE_OPTIONS=--openssl-legacy-provider
 ADD infra/package.json infra/yarn.lock /app/infra/
 COPY .yarn/ ./.yarn
 WORKDIR /app/infra
+RUN yarn config set npmRegistryServer "https://npm-upstream-821090935708.d.codeartifact.eu-west-1.amazonaws.com/npm/github-runner-npm-cache/"
```
Comment (busla) [inline infra/scripts/Dockerfile:13]: could we pass this as build-arg?

File: scripts/ci/Dockerfile [hunk 2]
```diff
@@ -4,13 +4,18 @@ ARG DOCKER_IMAGE_REGISTRY=public.ecr.aws
...
 
 WORKDIR /build
 
 # Adding and installing packages
 ADD package.json yarn.lock .yarnrc.yml ./
 COPY .yarn/ ./.yarn
 RUN apk add --update --no-cache python3 build-base gcc && ln -sf /usr/bin/python3 /usr/bin/python
+
+RUN yarn config set npmRegistryServer "https://npm-upstream-821090935708.d.codeartifact.eu-west-1.amazonaws.com/npm/github-runner-npm-cache/"
+RUN yarn config set 'npmRegistries["https://npm-upstream-821090935708.d.codeartifact.eu-west-1.amazonaws.com/npm/github-runner-npm-cache/"].npmAuthToken' `aws codeartifact get-authorization-token --domain npm-upstream --domain-owner 821090935708 --region eu-west-1 --query authorizationToken --output text`
```
Comment (busla) [inline scripts/ci/Dockerfile:17]: should  we maybe get the token before building the image and pass it in as docker secret?

File: scripts/ci/Dockerfile [hunk 3]
```diff
@@ -124,7 +129,12 @@ FROM mcr.microsoft.com/playwright:v${PLAYWRIGHT_VERSION}-focal as playwright-bas
 
 FROM playwright-base as output-playwright
 # TODO: remove awscli dependency (157 MB extra)
-RUN apt-get update && apt-get install -y zip awscli && apt-get purge
+RUN apt-get update && apt-get install -y zip && apt-get purge
+
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
```
Comment (busla) [inline scripts/ci/Dockerfile:134]: wondering if we should add a separate stage for the aws-cli and then copy the binary from that stage output

```dockerfile
FROM alpine3.15

ARG AWSCLI_VERSION="1.22.24"
ENV AWSCLI_VERSION=${AWSCLI_VERSION}

RUN apk add \
    bash \
    python3 \
    python3-dev \
    py-pip \
    build-base \
    jq \
    curl

RUN pip install awscli==$AWSCLI_VERSION --user
RUN mkdir -p /build

COPY /root/.local/bin/aws /build/
```

```dockerfile
...
COPY --chown=someuser:someuser --from=awscli /build /usr/local/bin
...
```

**Review by busla — CHANGES_REQUESTED** [review]
secret and multistage comment

File: scripts/ci/Dockerfile (hunk 2 above)
Comment (brynjarorng) [reply scripts/ci/Dockerfile:17]: Does that not leave a trace of the token in github CI uncovered?

File: scripts/ci/Dockerfile (hunk 3 above)
Comment (brynjarorng) [reply scripts/ci/Dockerfile]: Yep, good idea. Implemented

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Comment by andes-it** [comment]
This PR closed its doors. Sat around for a day after the stale tag, then reckoned it's not worth the fuss.

**Pull request closed without merging**